packages/flutter/lib/src/gestures/converter.dart

```diff
@@ -104,90 +93,44 @@ class PointerEventConverter {
           );
           break;
         case ui.PointerChange.hover:
```

Comment:
An assert should also trigger if the event we receive from the engine doesn't follow the expected sequence.

Reply:
i haven't looked to see if it makes sense here (i'll defer to @goderbauer) but as a general rule we have many places where we have debug-only code (using asserts to make it only visible in debug mode and free in release builds) to verify invariants like this.